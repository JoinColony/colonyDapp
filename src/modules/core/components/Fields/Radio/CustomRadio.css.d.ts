export const input: string;
export const main: string;
export const themePrimary: string;
export const themeDanger: string;
export const themeTraditional: string;
export const stateIsChecked: string;
export const description: string;
export const stateIsDisabled: string;
export const label: string;
export const icon: string;
export const content: string;
export const directionVertical: string;
export const customRadioIcon: string;
export const checkedCustomRadioIcon: string;
export const customRadioCheck: string;
