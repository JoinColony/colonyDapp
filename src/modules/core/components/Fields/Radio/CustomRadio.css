.input {
  height: 0;
  width: 0;
  opacity: 0;
}

.main {
  display: flex;
  align-items: center;
  padding: 7px;
  height: 41px;
  width: 100%;
  overflow: hidden;
  border: 1px solid transparent;
  border-radius: var(--radius-normal);
  background-color: color-mod(var(--temp-grey-blue-7) alpha(6%));
  color: var(--dark);
  cursor: pointer;
}

.themePrimary {
  composes: main;
}

.themeDanger {
  composes: main;
}

.themeGreyWithCircle {
  composes: main;
  border-color: color-mod(var(--action-secondary) alpha(25%));
  border-radius: var(--radius-tiny);
  background-color: color-mod(var(--temp-grey-blue-7) alpha(5%));
}

.themePrimary.stateIsChecked,
.themeGreyWithCircle.stateIsChecked {
  border: 1px solid var(--primary);
  background-color: color-mod(var(--primary) alpha(12%));
}

.themeDanger.stateIsChecked {
  border: 1px solid var(--pink);
  background-color: color-mod(var(--pink) alpha(12%));
}

.themePrimary:hover {
  background-color: color-mod(var(--primary) alpha(12%));
}

.themeDanger:hover {
  background-color: color-mod(var(--pink) alpha(12%));
}

.main.stateIsChecked .description {
  color: var(--temp-grey-blue-7);
  opacity: 0.85;
}

.main.stateIsDisabled .description {
  color: var(--temp-grey-blue-7);
  opacity: 0.85;
}

.main.stateIsDisabled .label {
  color: color-mod(var(--dark) alpha(65%));
}

.main.stateIsDisabled {
  border-color: transparent;
  background-color: var(--text-disabled);
  cursor: default;
}

.label {
  display: flex;
  align-items: center;
  font-size: var(--size-smallish);
  font-weight: var(--weight-bold);
  gap: 6px;
}

.description,
.description span {
  margin-left: auto;
  font-size: var(--size-tiny);
  font-weight: var(--weight-bold);
}

.icon {
  margin-right: 10px;
}

.content {
  display: flex;
  align-items: baseline;
  width: 100%;
  white-space: nowrap;
  gap: 6px;
}

.directionVertical .content {
  flex-direction: column;
  gap: 0;
}

.directionVertical .description {
  margin-left: 0;
}

.customRadioIcon {
  display: flex;
  flex-shrink: 0;
  justify-content: center;
  align-items: center;
  margin-right: 10px;
  height: 27px;
  width: 27px;
  border: 1px solid var(--temp-grey-blue-4);
  border-radius: 50%;
  background: var(--colony-white);
  box-shadow: inset 0px 2px 4px rgba(14, 37, 88, 0.07);
}

.checkedCustomRadioIcon {
  border-color: var(--primary);
}

.customRadioCheck {
  height: 7px;
  width: 7px;
  border-radius: 50%;
  background: var(--primary);
  box-shadow: var(--box-shadow-medium);
}
