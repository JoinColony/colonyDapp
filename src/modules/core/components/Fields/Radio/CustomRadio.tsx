import React, { useRef } from 'react';
import { MessageDescriptor, FormattedMessage } from 'react-intl';
import { nanoid } from 'nanoid';
import { useField } from 'formik';

import { getMainClasses } from '~utils/css';
import Icon from '~core/Icon';
import { UniversalMessageValues } from '~types/index';

import styles from './CustomRadio.css';

export interface Appearance {
  theme?: 'primary' | 'danger' | 'traditional';
  direction?: 'horizontal' | 'vertical';
}

export interface Props {
  /** Appearance object */
  appearance?: Appearance;
  /** Disable the input */
  disabled?: boolean;
  /** HTML input value */
  value: string;
  /** Label text */
  label: string | MessageDescriptor;
  /** Description text values for intl interpolation */
  labelValues?: UniversalMessageValues;
  /** Button description */
  description?: string | MessageDescriptor;
  /** Description text values for intl interpolation */
  descriptionValues?: UniversalMessageValues;
  /** Button icon */
  icon?: string;
  /** If the input is checked */
  checked: boolean;
  /** Radio button name attribute */
  name: string;
  /** Html input `id` attribute */
  inputId?: string;
}

const displayName = 'CustomRadio';

const CustomRadio = ({
  disabled,
  value,
  label,
  labelValues,
  checked,
  name,
  inputId,
  appearance = { theme: 'primary' },
  description,
  descriptionValues,
  icon,
}: Props) => {
  const [, { error }, { setValue }] = useField(name);
  const inputRef = useRef<string>(inputId || nanoid());

  return (
    <label
      className={getMainClasses(appearance, styles, {
        isChecked: checked,
        isDisabled: !!disabled,
      })}
      htmlFor={inputRef.current}
    >
      <input
        aria-checked={checked}
        aria-disabled={disabled}
        aria-invalid={!!error}
        disabled={disabled}
        type="radio"
        value={value}
        onClick={() => setValue(value)}
        name={name}
        id={inputRef.current}
        className={styles.input}
      />
      {icon && (
        <div className={styles.icon}>
          <Icon appearance={{ size: 'medium' }} name={icon} title={icon} />
        </div>
      )}
      <div className={styles.content}>
        {label && (
          <span className={styles.label}>
            {label === 'string' ? (
              label
            ) : (
              <FormattedMessage {...label} values={labelValues} />
            )}
          </span>
        )}
        {description && (
          <span className={styles.description}>
            {description === 'string' ? (
              description
            ) : (
              <FormattedMessage {...description} values={descriptionValues} />
            )}
          </span>
        )}
      </div>
    </label>
  );
};

CustomRadio.displayName = displayName;

export default CustomRadio;
