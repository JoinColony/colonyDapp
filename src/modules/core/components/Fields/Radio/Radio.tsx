import React, { ReactNode, useRef } from 'react';
import nanoid from 'nanoid';

import { getMainClasses } from '~utils/css';

import InputLabel from '../InputLabel';
import asField from '../asField';
import { AsFieldEnhancedProps } from '../types';

import styles from './Radio.css';

export interface Appearance {
  direction?: 'horizontal' | 'vertical';
  theme?: 'buttonGroup' | 'fakeCheckbox' | 'colorPicker';
}

interface Props {
  /** Appearance object */
  appearance?: Appearance;
  /** If the input is checked */
  checked: boolean;
  /** Children to render in place of the default label */
  children?: ReactNode;
  /** Disable the input */
  disabled?: boolean;
  /** Style object for the visible radio */
  radioStyle?: { [k: string]: string };
  inputId?: string;
  value?: any;
}

const displayName = 'Radio';

const Radio = ({
  appearance = {
    direction: 'vertical',
  },
  checked,
  children,
  disabled = false,
  elementOnly = false,
  $error,
  help,
  inputId: inputIdProp,
  label,
  name,
  $value,
  radioStyle,
  /* eslint-disable @typescript-eslint/no-unused-vars */
  connect,
  $id,
  formatIntl,
  $touched,
  setError,
  setValue,
  isSubmitting,
  /* eslint-enable @typescript-eslint/no-unused-vars */
  ...props
}: Props & AsFieldEnhancedProps) => {
  const { current: inputId } = useRef<string>(inputIdProp || nanoid());
  return (
    <label
      className={getMainClasses(appearance, styles, {
        customChildren: !!children,
        isChecked: checked,
        isDisabled: !!disabled,
      })}
      htmlFor={elementOnly ? inputId : undefined}
    >
      <>
        <input
          className={styles.delegate}
          value={$value}
          name={name}
          type="radio"
          id={inputId}
          disabled={disabled}
          aria-checked={checked}
          aria-disabled={disabled}
          aria-invalid={!!$error}
          {...props}
        />
        <span className={styles.radio} style={radioStyle}>
          {!!appearance && appearance.theme === 'fakeCheckbox' && (
            <span className={styles.checkmark} />
          )}
        </span>
        {!elementOnly && !!label ? (
          <span className={styles.labelContainer}>
            <InputLabel
              appearance={{ direction: 'horizontal' }}
              label={label}
              help={help}
              inputId={inputId}
            />
          </span>
        ) : (
          label || children
        )}
      </>
    </label>
  );
};

Radio.displayName = displayName;

export default asField<Props>()(Radio);
