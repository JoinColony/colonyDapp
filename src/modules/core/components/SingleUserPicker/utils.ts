import { AnyUser } from '~data/index';

/*
Extracts the required values to be used in the SingleUserPicker
on selection
*/
export const filterUserSelection = (data: AnyUser[], filterValue: string) => {
  if (!filterValue) {
    return data;
  }

  const filteredUsers = data.filter((user) => {
    const username = user.profile?.username || '';
    const walletAddress = user.profile?.walletAddress || '';

    return (
      username.toLowerCase().includes(filterValue.toLowerCase()) ||
      walletAddress.toLowerCase().includes(filterValue.toLowerCase()) ||
      user.id.toLowerCase().includes(filterValue.toLowerCase())
    );
  });

  const customUserValue: AnyUser = {
    id: 'filterValue',
    profile: {
      walletAddress: filterValue,
      displayName: filterValue,
    },
  };

  return [customUserValue].concat(filteredUsers);
};
