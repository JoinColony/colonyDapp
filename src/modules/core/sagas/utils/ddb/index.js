/* @flow */

import getDDB from './getDDB';

// eslint-disable-next-line import/prefer-default-export
export { getDDB };
