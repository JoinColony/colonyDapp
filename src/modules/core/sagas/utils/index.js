/* @flow */

import getDDB from './getDDB';
import getNetworkClient from './getNetworkClient';

export * from './transactions';
export { getDDB, getNetworkClient };
