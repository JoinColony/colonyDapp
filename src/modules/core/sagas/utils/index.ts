export { default as getColonyManager } from './getColonyManager';
export { default as getGasPrices } from './getGasPrices';
export { getMethod, getTransactionMethod } from './getMethod';
export { default as getWalletCategory } from './getWalletCategory';
