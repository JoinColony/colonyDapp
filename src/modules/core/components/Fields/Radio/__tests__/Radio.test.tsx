/* eslint-env jest */

import React from 'react';
import { Formik } from 'formik';
import toJson from 'enzyme-to-json';

import { mountWithIntl } from '../../../../../../__tests__/utils';

import Radio from '../Radio';

describe('Radio component', () => {
  test('Renders initial component', () => {
    const wrapper = mountWithIntl(
      <Formik initialValues={{}} onSubmit={() => undefined}>
        <Radio
          checked={false}
          name="radioInput"
          help="halp"
          inputId="halpHalp"
          label="awesome"
          value="radioInput"
        />
      </Formik>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
