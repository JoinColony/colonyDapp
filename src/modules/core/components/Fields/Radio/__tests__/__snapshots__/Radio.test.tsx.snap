// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio component Renders initial component 1`] = `
<Formik
  initialValues={Object {}}
  onSubmit={[Function]}
>
  <injectIntl(Component)
    checked={false}
    help="halp"
    inputId="halpHalp"
    label="awesome"
    name="radioInput"
    value="radioInput"
  >
    <Component
      checked={false}
      help="halp"
      inputId="halpHalp"
      intl={
        Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatDateToParts": [Function],
          "formatHTMLMessage": [Function],
          "formatList": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatNumberToParts": [Function],
          "formatPlural": [Function],
          "formatRelativeTime": [Function],
          "formatTime": [Function],
          "formatTimeToParts": [Function],
          "formats": Object {},
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getListFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralRules": [Function],
            "getRelativeTimeFormat": [Function],
          },
          "locale": "en",
          "messages": Object {},
          "onError": [Function],
          "textComponent": Symbol(react.fragment),
          "timeZone": undefined,
        }
      }
      label="awesome"
      name="radioInput"
      value="radioInput"
    >
      <Field
        checked={false}
        component={[Function]}
        connect={true}
        help="halp"
        inputId="halpHalp"
        intl={
          Object {
            "defaultFormats": Object {},
            "defaultLocale": "en",
            "formatDate": [Function],
            "formatDateToParts": [Function],
            "formatHTMLMessage": [Function],
            "formatList": [Function],
            "formatMessage": [Function],
            "formatNumber": [Function],
            "formatNumberToParts": [Function],
            "formatPlural": [Function],
            "formatRelativeTime": [Function],
            "formatTime": [Function],
            "formatTimeToParts": [Function],
            "formats": Object {},
            "formatters": Object {
              "getDateTimeFormat": [Function],
              "getListFormat": [Function],
              "getMessageFormat": [Function],
              "getNumberFormat": [Function],
              "getPluralRules": [Function],
              "getRelativeTimeFormat": [Function],
            },
            "locale": "en",
            "messages": Object {},
            "onError": [Function],
            "textComponent": Symbol(react.fragment),
            "timeZone": undefined,
          }
        }
        label="awesome"
        name="radioInput"
        value="radioInput"
      >
        <mapProps(Radio)
          checked={false}
          connect={true}
          field={
            Object {
              "name": "radioInput",
              "onBlur": [Function],
              "onChange": [Function],
              "value": undefined,
            }
          }
          form={
            Object {
              "dirty": false,
              "errors": Object {},
              "getFieldMeta": [Function],
              "getFieldProps": [Function],
              "handleBlur": [Function],
              "handleChange": [Function],
              "handleReset": [Function],
              "handleSubmit": [Function],
              "initialErrors": Object {},
              "initialStatus": undefined,
              "initialTouched": Object {},
              "initialValues": Object {},
              "isSubmitting": false,
              "isValid": true,
              "isValidating": false,
              "registerField": [Function],
              "resetForm": [Function],
              "setErrors": [Function],
              "setFieldError": [Function],
              "setFieldTouched": [Function],
              "setFieldValue": [Function],
              "setFormikState": [Function],
              "setStatus": [Function],
              "setSubmitting": [Function],
              "setTouched": [Function],
              "setValues": [Function],
              "status": undefined,
              "submitCount": 0,
              "submitForm": [Function],
              "touched": Object {},
              "unregisterField": [Function],
              "validateField": [Function],
              "validateForm": [Function],
              "validateOnBlur": true,
              "validateOnChange": true,
              "validateOnMount": false,
              "values": Object {},
            }
          }
          help="halp"
          inputId="halpHalp"
          intl={
            Object {
              "defaultFormats": Object {},
              "defaultLocale": "en",
              "formatDate": [Function],
              "formatDateToParts": [Function],
              "formatHTMLMessage": [Function],
              "formatList": [Function],
              "formatMessage": [Function],
              "formatNumber": [Function],
              "formatNumberToParts": [Function],
              "formatPlural": [Function],
              "formatRelativeTime": [Function],
              "formatTime": [Function],
              "formatTimeToParts": [Function],
              "formats": Object {},
              "formatters": Object {
                "getDateTimeFormat": [Function],
                "getListFormat": [Function],
                "getMessageFormat": [Function],
                "getNumberFormat": [Function],
                "getPluralRules": [Function],
                "getRelativeTimeFormat": [Function],
              },
              "locale": "en",
              "messages": Object {},
              "onError": [Function],
              "textComponent": Symbol(react.fragment),
              "timeZone": undefined,
            }
          }
          label="awesome"
          value="radioInput"
        >
          <Radio
            $id="radioInput"
            aria-invalid={false}
            aria-label="awesome"
            aria-labelledby="radioInput-label"
            checked={false}
            connect={true}
            formatIntl={[Function]}
            help="halp"
            inputId="halpHalp"
            isSubmitting={false}
            label="awesome"
            name="radioInput"
            onBlur={[Function]}
            onChange={[Function]}
            setError={[Function]}
            setValue={[Function]}
            title="awesome"
            value="radioInput"
          >
            <label
              className=""
            >
              <input
                aria-checked={false}
                aria-disabled={false}
                aria-invalid={false}
                aria-label="awesome"
                aria-labelledby="radioInput-label"
                disabled={false}
                id="halpHalp"
                name="radioInput"
                onBlur={[Function]}
                onChange={[Function]}
                title="awesome"
                type="radio"
                value="radioInput"
              />
              <span />
              <span>
                <InputLabel
                  appearance={
                    Object {
                      "direction": "horizontal",
                    }
                  }
                  help="halp"
                  inputId="halpHalp"
                  label="awesome"
                >
                  <label
                    className=""
                    htmlFor="halpHalp"
                    id="halpHalp-label"
                  >
                    <span>
                      awesome
                    </span>
                    <span>
                      halp
                    </span>
                  </label>
                </InputLabel>
              </span>
            </label>
          </Radio>
        </mapProps(Radio)>
      </Field>
    </Component>
  </injectIntl(Component)>
</Formik>
`;
